cdap-watchdog/src/main/java/co/cask/cdap/logging/plugins/LocationManager.java

```diff
@@ -122,7 +126,8 @@ OutputStream getLocationOutputStream(LocationIdentifier locationIdentifier, Stri
         // create new file and open outputstream on it
         logFile.createNew(filePermissions);
         // TODO: Handle existing file in a better way rather than copying it over
-        OutputStream outputStream = new LocationOutputStream(logFile, logFile.getOutputStream());
+        OutputStream outputStream = new LocationOutputStream(logFile, logFile.getOutputStream(),
```

Comment:
we also need to pass permissions when we do getOutputStream. look at https://github.com/caskdata/cdap/pull/8044